pypeit/core/telluric.py

```diff
@@ -135,22 +134,73 @@ def pca_eval(theta,pca_dict):
 #        theta_pca[2:npca+1]``
 #
 #    """
-#    gaussian_mixture_model = pca_dict['prior']
+#    gaussian_mixture_model = qso_pca_dict['prior']
 #    A = theta[2:]
 #    return gaussian_mixture_model.score_samples(A.reshape(1,-1))
 
 
+def read_telluric_pca(filename, wave_min=None, wave_max=None, pad_frac=0.10):
+    """
+    Reads in the telluric PCA components from a file.
+
+    Optionally, this method also trims wavelength to be in within ``wave_min``
+    and ``wave_max`` and pads the data (see ``pad_frac``).
+
+    Args:
+        filename (:obj:`str`):
+            Telluric PCA filename
+        wave_min (:obj:`float`):
+            Minimum wavelength at which the grid is desired
+        wave_max (:obj:`float`):
+            Maximum wavelength at which the grid is desired.
+        pad_frac (:obj:`float`):
+            Percentage padding to be added to the grid boundaries if
+            ``wave_min`` or ``wave_max`` are input; ignored otherwise. The
+            resulting grid will extend from ``(1.0 - pad_frac)*wave_min`` to
+            ``(1.0 + pad_frac)*wave_max``.
+
+    Returns:
+        :obj:`dict`: Dictionary containing the telluric PCA components.
+            - wave_grid: Telluric model wavelength grid
+            - dloglam: Wavelength sampling of telluric model
+            - tell_pad_pix: Number of pixels to pad at edges for convolution
+            - ncomp_tell_pca: Number of PCA components
+            - tell_pca: PCA component vectors
+            - bounds_tell_pca: Maximum/minimum coefficient
+            - coefs_tell_pca: Set of model coefficient values (for prior in future)
+            - teltype: Type of telluric model, i.e. 'PCA'
+    """
+    # load_telluric_grid() takes care of path and existance check
+    hdul = data.load_telluric_grid(filename)
+    wave_grid_full = hdul[1].data
+    pca_comp_full = hdul[0].data
+    ncomp = hdul[0].header['NCOMP']
+    bounds = hdul[2].data
+    model_coefs = hdul[3].data
+
+    ind_lower = np.argmin(np.abs(wave_grid_full - (1.0 - pad_frac)*wave_min)) \
+                    if wave_min is not None else 0
+    ind_upper = np.argmin(np.abs(wave_grid_full - (1.0 + pad_frac)*wave_max)) \
+                    if wave_max is not None else nspec_full
```

Comment:
Is `nspec_full` defined? Should this instead be the maximum valid wavelength of the models?

Reply:
Oops -- this was a carryover from `read_telluric_grid` but I forgot to copy over its definition. Same response as above applies.